File: Python/Count_Binary_Substrings.py
# 696. Count Binary Substrings

def countBinarySubstrings(self, s: str) -> int:
    current_count = 1
    previous_count = 0
    ans = 0
    for i in range(1, len(s)):
        if s[i] == s[i - 1]:
            current_count += 1
        else:
            ans += min(current_count, previous_count)
            previous_count = current_count
            current_count = 1

    return ans + min(current_count, previous_count)

File: JavaScript/count-binary-substrings.js
// 696. Count Binary Substrings

var countBinarySubstrings = function(s) {
    let currentCount = 1;
    let prevCount = 0;
    let ans = 0;
    for (let i = 1; i < s.length; i++)
        if (s[i] === s[i-1]) 
            currentCount++;
        else {
            ans += Math.min(currentCount, prevCount);
            prevCount = currentCount;
            currentCount = 1;
        }
    return ans + Math.min(currentCount, prevCount)
};
